src/test/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheckTest.java

```diff
@@ -290,4 +294,59 @@ public void testIgnoreCommentsCppStyleWithIllegalPatternFalse() throws Exception
         verify(checkConfig, getPath("InputRegexpTrailingComment11.java"), expected);
     }
 
+    @Test
+    public void testStateIsClearedOnBeginTreeErrorCount() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(RegexpCheck.class);
+        checkConfig.addProperty("format", "^import");
+        checkConfig.addProperty("errorLimit", "2");
+        checkConfig.addProperty("duplicateLimit", "1");
```

Comment:
What blocks us to use inline in Input config?
Input files should not be reused with other test methods.

Reply:
If you are asking for InlineConfigParser usage in beginTree test then right know it is not possible to cover with that.

2) for this particular check even thought if i write config above the file it then output will be affect like https://github.com/checkstyle/checkstyle/pull/13367#discussion_r1257769744

Reply:
Sorry but I do not understand your comment. I read it several times and still no way to understand it.
You did multiple coverages for multiple files before, it should work.
If I miss something, please make changes (may me separate  PR) and  point to a problem by log or screenshot

Reply:
I was trying to say that inline config parser is not able to run 2 files sequentially, which may be(Not sure) possible if only 1 file contains violation and one not or may be both of them not. in this case it may be possible that we use inline config parser.

but for this check my suggestion is to use only verify the reason is both the file contains violation

Reply:
@romani sorry,
https://github.com/checkstyle/checkstyle/pull/13383#discussion_r1264112474 but this is same in this we have not set any extra property.

see this pr
https://github.com/checkstyle/checkstyle/commit/608599c9a775e468a13b7d85da20a292bcf38509#diff-5afa2cb9023b6fdb95b12684cd82f31c9092c153cdf21674d94feeef820ae122R247

Reply:
@romani please look https://github.com/checkstyle/checkstyle/pull/13383#discussion_r1264724795

Reply:
What blocks put this extra property to Input config?

Reply:
@Kevin222004 , please put properties to Input file, if you have problem to do this - please share.

Reply:
@romani sorry I have not visited this pr so long.

First thing we don't have any method in existing code verifyWithInlineConfigParser.

I have created a new one verifyWithInlineConfigParser